orbit_predictor/predictors/pass_iterators.py

```diff
@@ -201,84 +201,68 @@ def _find_los(self, tca):
 class SmartLocationPredictor(BaseLocationPredictor):
 
     def iter_passes(self):
-        current_date = self.start_date
-        while True:
-            valid, aos, tca, los, max_elevation = self._find_next_pass(current_date)
-            current_date = los
+        # Explore all values of t every 3 minutes
+        t_values = list(range(0, int((self.limit_date - self.start_date).total_seconds()), 180))
```

Comment:
This is a big array of integers, I wouldn't use plain list. We could use `array`, from [array library](https://docs.python.org/3/library/array.html). Something like:
`t_values = array.array('L', range(0, int((self.limit_date - self.start_date).total_seconds()), 180))`

Reply:
Good point. For some reason I didn't want to import NumPy, but we are already installing it so it's not a big deal. I'll change it.

Reply:
Not `numpy` array, the `array` library from the standard  ;)

Reply:
Done!